### PR TITLE
fix: mysql-2: support extract values in all known calling conventions

### PR DESCRIPTION
**Pull request by mmanciop**
Collect the values passed as named or unnamed parameters in to `mysql-2`'s `.query()` and `.execute()` for the following calling conventions:

* Values nested in `options` object: `client.query({ sql: 'SELECT * FROM users WHERE id = ?', values: ['123']}, callback?)`
* Single value: `client.query('SELECT * FROM users WHERE id = ?', '123', callback?)`
* Named parameters: `client.query('SELECT * FROM users WHERE id = ?id', {id: '123'}, callback?)`

### CODE REVIEW

**Review by doriaviram — APPROVED** [review]

File: src/hooks/mySql.js [hunk 1]
```diff
@@ -32,11 +32,26 @@ function extractQueryFromArg(arg) {
   }
 }
 
+function extractValuesFromArg(arg) {
+  if (isObject(arg) && typeof arg === 'object' && !!arg['values']) {
+    return arg['values'];
+  }
+}
```
Comment (therightstuff) [inline src/hooks/mySql.js:39]: please add an explicit return, even if it's null/undefined

File: src/hooks/mySql.js [hunk 2]
```diff
@@ -32,11 +32,26 @@ function extractQueryFromArg(arg) {
...
+
 function queryBeforeHook(args, extenderContext) {
   const awsRequestId = TracerGlobals.getHandlerInputs().context.awsRequestId;
   const transactionId = getCurrentTransactionId();
   const query = extractQueryFromArg(args[0]);
-  const values = Array.isArray(args[1]) ? args[1] : [];
+
+  const values =
+    // This is for the `.query(options, callback?)` form
+    extractValuesFromArg(args[0]) ||
```
Comment (therightstuff) [inline src/hooks/mySql.js:48]: i would argue that this method should be named `extractValuesFromQueryOptions` and that the second "half" of the ternary be extracted to a method named `extractValuesFromQueryParameter`

File: src/hooks/mySql.test.js [hunk 3]
```diff
@@ -52,279 +52,338 @@ describe('mySql', () => {
...
-      expect(spans).toEqual([
-        createBaseBuilderFromSpan(spans[0])
-          .withQuery('Unknown')
-          .withConnectionParameters(DUMMY_OPTIONS)
-          .withResponse(createExpectedResponse())
-          .build(),
-      ]);
-      done();
+    test('hook -> query (text: string, callback: Function) -> fail', (done) => {
+      const error = new Error('DuumyError');
```
Comment (therightstuff) [inline src/hooks/mySql.test.js:90]: trivial typo: DummyError (copy/pasted 12 times)

**Review by therightstuff — CHANGES_REQUESTED** [review]
looking good! one minor request and one trivial observation, but overall happiness

File: src/hooks/mySql.js (hunk 2 above)
Comment (mmanciop) [reply src/hooks/mySql.js]: Fixed in 5fbb948901e88348c95947aa27909090528b9fde

File: src/hooks/mySql.js (hunk 1 above)
Comment (mmanciop) [reply src/hooks/mySql.js:51]: Fixed in dc7a11fbf24d18f3fa4ca6f207ead57ec93b0910

File: src/hooks/mySql.test.js (hunk 3 above)
Comment (mmanciop) [reply src/hooks/mySql.test.js]: Fixed in b1086a53a98675ac5ccae59c5c31b5303d8d404d

**Review by therightstuff — APPROVED** [review]
excellent! thank you for the fixes :)

**Pull request merged**